File: Tutorial5-3Dagger-Android-MutipleActivitiesAndFragments/src/main/java/com/example/tutorial5_3dagger_android_mutipleactivitiesandfragments/MyApplication.java
package com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments;

import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.di.AppComponent;
import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.di.DaggerAppComponent;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;

public class MyApplication extends DaggerApplication {

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        AppComponent appComponent = DaggerAppComponent.builder().application(this).build();
        return appComponent;
    }

}

File: Tutorial8-3DaggerHilt-DefineComponent/src/main/java/com/smarttoolfactory/tutorial8_3daggerhilt_definecomponent/MyApplication.kt
package com.smarttoolfactory.tutorial8_3daggerhilt_definecomponent

import android.app.Application
import com.smarttoolfactory.tutorial8_3daggerhilt_definecomponent.di.AppComponent
import dagger.hilt.android.HiltAndroidApp
import javax.inject.Provider

@HiltAndroidApp
class MyApplication : Application() {

//    private val appComponentProvider by lazy {
//        Provider<AppComponent.Builder> {
//
//        }
//    }

    val appComponent by lazy {
//        appComponentProvider.get().application(this).build()

    }

}